arvo/eyre/examples.md

```diff
@@ -0,0 +1,689 @@
+
+Here's a very simple generator that will just echo back the body of the request (if available) along with the current datetime. You can save it in the `/gen` directory and `|commit %home`.
+
+`eyre-gen.hoon`
+
+```hoon
+|=  [[now=@da eny=@uvJ bec=beak] ~ ~]
+|=  [authenticated=? =request:http]
+^-  simple-payload:http
+=/  msg=@t
+  ?~  body.request
+    (scot %da now)
+  (cat 3 (cat 3 (scot %da now) 10) q.u.body.request)
+=/  data=octs
+  (as-octs:mimes:html msg)
+=/  content-length=@t
+  (rsh [3 2] (scot %ui p.data))
+=/  =response-header:http
+  :-  200
+  :~  ['Content-Length' content-length]
+      ['Content-Type' 'text/plain']
+  ==
+[response-header `data]
+```
+
+Eyre requires generators to be a gate within a gate. The sample of the first gate must be:
+
+```hoon
+[[now=@da eny=@uvJ bec=beak] ~ ~]
+```
+
+The sample of the second nested gate must be:
+
+```hoon
+[authenticated=? =request:http]
+```
+
+The return type of the generator must be [simple-payload:http](@/docs/arvo/eyre/data-types.md#simple-payload-http).
+
+If you look at our example generator you'll see it meets these requirements.
+
+In order to make our generator available, we must bind it to a URL path. To do this, we send Eyre a `%serve` `task`, which looks like:
+
+```hoon
+[%serve =binding =generator]
+```
+
+The [binding](@/docs/arvo/eyre/data-types.md#binding) specifies the site and URL path, and the [generator](@/docs/arvo/eyre/data-types.md#generator) specifies the `desk`, the `path` to the generator, and arguments. Note that the passing of arguments to the generator by Eyre is not currently implemented, so you can just leave that as `~` since it won't do anything.
+
+Let's bind our generator to the `/mygen` URL path with the `|pass` command in the dojo:
+
+```
+|pass [%e [%serve `/mygen %home /gen/eyre-gen/hoon ~]]
+```
+
+Note that Eyre responds with a `%bound` `gift` to indicate whether the binding succeeded but `|pass` doesn't take such responses so it's not shown.
+
+Now let's try making an HTTP request using `curl` in the unix terminal:
+
+```
+curl -i http://localhost:8080/mygen --data 'blah blah blah'
+```
+
+We can see that the request has succeeded and our generator has responded with the datetime and request body:
+
+```
+HTTP/1.1 200 ok
+Date: Sat, 29 May 2021 09:19:45 GMT
+Connection: keep-alive
+Content-Length: 41
+Server: urbit/vere-1.5
+Content-Type: text/plain
+Content-Length: 41
+
+~2021.5.29..09.19.45..e096
+blah blah blah
+```
+
+# Managing CORS Origins
+
+Here we'll look at approving and rejecting a CORS origin by passing Clay a [%approve-origin](@/docs/arvo/eyre/tasks.md#approve-origin) `task` and [%reject-origin](@/docs/arvo/eyre/tasks.md#reject-origin) `task` respectively.
+
+First, using `|pass` in the dojo, let's approve the origin `http://foo.example` by sending Eyre a `%approve-origin` `task`:
```

Comment:
Any reason for sticking with `|pass` here instead of the manual task & scry construction? If it's for educational purposes, we should probably at least mention the generators as the thing you'll want for every-day use.